plc4j/protocols/driver-bases/base/src/main/java/org/apache/plc4x/java/base/connection/AbstractPlcConnection.java

```diff
@@ -57,6 +67,42 @@ public boolean canSubscribe() {
         return false;
     }
 
+    /**
+     * The default implementation uses the {@link #ping(int)} method.
+     * Drivers that want to implement a more specific version have to overide it.
+     */
+    @Override
+    public boolean isConnected() {
+        return ping(PING_TIMEOUT_MS);
+    }
+
+    /**
+     * Method that sends a test-request or ping to the PLC to check if the PLC is still there.
+     * In most cases this method should only be used from the {@link #isConnected()} method.
+     * This method can only be used if ghe {@link #getInetSocketAddress()} returns a Socket Adress.
+     * Otherwise it throws a {@link NotImplementedException} to inform the user about that.
+     */
+    protected boolean ping(int timeout) {
+        Optional<InetSocketAddress> address = getInetSocketAddress();
+        if (!address.isPresent()) {
+            throw new NotImplementedException("Tries to check the connection with ping, " +
+                "but no Socket Address is given!");
+        }
+        try (Socket s = new Socket()) {
+            s.connect(address.get(), timeout);
```

Comment:
I'm not sure if we should call .disconnect() here as well. Does socket.close() does this. If thats isn't the case we should call it so the socket doesn't stay open on the other side

Reply:
and the disconnect would hurt. What we could do though is using SO_REUSEADDR (if that applies for clients at all) to speed things up

Reply:
Just checked the docs (https://docs.oracle.com/javase/7/docs/technotes/guides/net/articles/connection_release.html) and it seems that #close() does all thats necessary.
I added the SO_REUSEADDR but I'm unsure ablout it. Maybe others have an opinion about that?

Reply:
might only apply to ServerSockets so we should be good here: https://docs.oracle.com/javase/7/docs/api/java/net/StandardSocketOptions.html#SO_REUSEADDR